lib/Parser/Scan.cpp

```diff
@@ -2087,36 +2093,11 @@ tokens Scanner<EncodingPolicy>::ScanCore(bool identifyKwds)
             case '-':
                 p++;
                 token = tkDec;
-                if (m_fHtmlComments)
+                if (!m_fIsModuleCode)
                 {
-                    int i = 0;
-                    while ('-' == PeekFirst(p + i, last)) //Have already seen --, skip any further - characters
-                        i++;
-                    if ('>' == PeekFirst(p + i++, last)) //This means we've got a --------------------------->.
```

Comment:
There is a test for this, but GitHub is cutting it off in the diff. You can see it if you view the whole file.

I'll look into what other browsers do.